app/web/pages/_app.tsx

```diff
@@ -43,15 +49,82 @@ function MyApp({ Component, pageProps }: AppWithLayoutProps) {
     }
   }, []);
 
+  useEffect(() => {
+    const registerServiceWorker = async () => {
+      try {
+        const registration = await navigator.serviceWorker.register(
+          "/service-worker.js",
+          { scope: "/" }
+        );
+        console.log(
+          "Service Worker registered with scope:",
+          registration.scope
+        );
+
+        const existingPushSubscription =
+          await registration.pushManager.getSubscription();
+        const p256dhKey = existingPushSubscription?.getKey("p256dh");
+
+        if (existingPushSubscription && p256dhKey) {
+          const publicKey = arrayBufferToBase64(p256dhKey);
+          if (publicKey !== process.env.NEXT_PUBLIC_VAPID_PUBLIC_KEY) {
+            await existingPushSubscription.unsubscribe();
+          } else {
+            return;
+          }
+        }
+
+        const { vapidPublicKey } = await getVapidPublicKey();
+        const subscription: PushSubscription =
+          await registration.pushManager.subscribe({
+            userVisibleOnly: true,
+            applicationServerKey: vapidPublicKey,
+          });
+
+        await registerPushNotification(subscription);
```

Comment:
I think we should rename this function, right now it makes no sense. How about `regsiterPushNotificationSubscription`?